runtimes/peregrine/src/system/mod.rs

```diff
@@ -53,6 +53,12 @@ pub const SS_58_PREFIX: u16 = 38;
 /// Currency Symbol
 pub const CURRENCY_SYMBOL: &[u8] = b"PILT";
 
+/// The denomination of native currency.
+pub const DENOMINATION: u8 = 15;
```

Comment:
Why do we still need these values here if we now have the `InspectMetadata` trait?

Reply:
I am using these values for the implementation of the `InspectMetadata` trait. I could define them directly within the implementation, but I kinda like the const definition with a declarative name and documentation. Additionally, I am using these values in the parachain node implementation for generating a new chain spec.

Reply:
I think it would be best to have a single source of truth. You could require the trait in the node implementation directly, or then get rid of the trait and use constants everywhere, but not both.

Reply:
[19762e4](https://github.com/KILTprotocol/kilt-node/pull/818/commits/19762e4bb89b367442a29cb4c809c5c394764468)